src/openrct2/scripting/bindings/world/ScScenario.hpp

```diff
@@ -37,131 +37,149 @@ namespace OpenRCT2::Scripting
         { "finish5Rollercoasters", OBJECTIVE_FINISH_5_ROLLERCOASTERS },
         { "repayLoanAndParkValue", OBJECTIVE_REPAY_LOAN_AND_PARK_VALUE },
         { "monthlyFoodIncome", OBJECTIVE_MONTHLY_FOOD_INCOME },
+        { "modularSystem", OBJECTIVE_MODULAR_SYSTEM_V1 },
     });
 
     class ScScenarioObjective
     {
     private:
         std::string type_get()
         {
-            return std::string(ScenarioObjectiveTypeMap[gScenarioObjective.Type]);
+            return std::string(ScenarioObjectiveTypeMap[gScenarioObjective.LegacyType]);
         }
 
         void type_set(const std::string& value)
         {
             ThrowIfGameStateNotMutable();
-            gScenarioObjective.Type = ScenarioObjectiveTypeMap[value];
+            std::string details = gScenarioDetails;
+            gScenarioObjective.LegacyType = ScenarioObjectiveTypeMap[value];
+            gScenarioObjective.Reset();
+            gScenarioObjective.ConvertObjective(ScenarioObjectiveTypeMap[value], gScenarioDetails);
         }
 
         uint16_t guests_get()
         {
-            if (gScenarioObjective.Type == OBJECTIVE_GUESTS_BY || gScenarioObjective.Type == OBJECTIVE_GUESTS_AND_RATING)
+            if (gScenarioObjective.LegacyType == OBJECTIVE_GUESTS_BY
+                || gScenarioObjective.LegacyType == OBJECTIVE_GUESTS_AND_RATING)
             {
-                return gScenarioObjective.NumGuests;
+                return std::static_pointer_cast<ObjectiveGuestNumGoal>(gScenarioObjective.PhasedGoals[0].goals[0])
+                    ->GetGuestNumGoal();
             }
             return 0;
         }
 
         void guests_set(uint16_t value)
         {
             ThrowIfGameStateNotMutable();
-            if (gScenarioObjective.Type == OBJECTIVE_GUESTS_BY || gScenarioObjective.Type == OBJECTIVE_GUESTS_AND_RATING)
+            if (gScenarioObjective.LegacyType == OBJECTIVE_GUESTS_BY
+                || gScenarioObjective.LegacyType == OBJECTIVE_GUESTS_AND_RATING)
             {
-                gScenarioObjective.NumGuests = value;
+                return std::static_pointer_cast<ObjectiveGuestNumGoal>(gScenarioObjective.PhasedGoals[0].goals[0])
+                    ->SetGuestNumGoal(value);
             }
         }
 
         uint8_t year_get()
         {
-            if (gScenarioObjective.Type == OBJECTIVE_GUESTS_BY || gScenarioObjective.Type == OBJECTIVE_PARK_VALUE_BY)
+            if (gScenarioObjective.LegacyType == OBJECTIVE_GUESTS_BY
+                || gScenarioObjective.LegacyType == OBJECTIVE_PARK_VALUE_BY)
             {
-                return gScenarioObjective.Year;
+                return gScenarioObjective.PhasedGoals[0].yearDate;
```

Comment:
would it be possible to also be able to set the month? and also setting the minimum park rating would be cool

Reply:
Not through the plugin interface. Not yet at least. The plugin interface needs to remain the same for compatibility with older plugins. Sometime later I could write a new interface to use the new system, where months and minimum values are indeed supported.